lib/internal/modules/cjs/loader.js

```diff
@@ -756,7 +754,7 @@ Module.prototype._compile = function(content, filename) {
       undefined,
       false,
       undefined,
-      [],
+      [NativeModule.cjsContext],
```

Comment:
IIUC this works by nesting the scope with a `with` scope, which may result in more runtime calls when generating the code for access. Can you verify the performance impact compared to setting it on the global proxy?

Reply:
I've included some perf results below.